client/src/app/members/new-edit/member-new-edit.component.ts

```diff
@@ -60,44 +70,42 @@ export class MemberNewEditComponent implements OnInit {
       );
   }
 
-  initForm(): void {
-    this.registrationForm = this.formBuilder.group({
-      lastName: ['', Validators.required],
-      firstName: ['', Validators.required],
-      school: ['', Validators.required],
-      statuts: [false, ValidateCheckbox],
-      ri: [false, ValidateCheckbox],
-    });
-  }
-
-  initNameValidation(): void {
+  private initNameFormatter(): void {
     const { lastName, firstName } = this.registrationForm.controls;
 
-    lastName.valueChanges.subscribe(value => lastName.setValue(this.nameFormatter(value)));
-    firstName.valueChanges.subscribe(value => firstName.setValue(this.nameFormatter(value)));
-  }
-
-  async initEdit(): Promise<void> {
-    const { id } = await this.route.params.toPromise();
-
-    if (!id) return;
+    const nameFormatter = (name: string = ''): string =>
+      name
+        .replace(/[\wÀ-ÿ]+(\S&-)*/g, txt => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase())
+        .replace(/[`~!@#$%^&*()_|+\=?;:'",.<>\{\}\[\]\\\/]/gi, '');
 
-    const { lastName, firstName, school, statuts, ri } = this.registrationForm.controls;
-    this.memberId = +id;
-    this.memberService.getById(+this.memberId).subscribe((member) => {
-      lastName.setValue(member.lastName);
-      firstName.setValue(member.firstName);
-      school.setValue(member.school);
-      statuts.setValue(true);
-      ri.setValue(true);
-    });
+    lastName.valueChanges.subscribe(value => lastName.setValue(nameFormatter(value)));
+    firstName.valueChanges.subscribe(value => firstName.setValue(nameFormatter(value)));
   }
 
-  nameFormatter(name: string): string {
-    if (!name) return '';
-    return name
-      .replace(/[\wÀ-ÿ]+(\S&-)*/g, txt => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase())
-      .replace(/[`~!@#$%^&*()_|+\=?;:'",.<>\{\}\[\]\\\/]/gi, '');
+  private initEdit(): void {
+    const { lastName, firstName, school, statuts, ri } = this.registrationForm.controls;
+    console.log('DEBUG1');
+
+    this.route.paramMap.subscribe(_ => console.log('DEBUG2'));
+    this.route.params.subscribe(_ => console.log('DEBUG3'));
+    this.route.params.pipe(tap(_ => console.log('DEBUG4')));
+    this.route.queryParamMap.subscribe(_ => console.log('DEBUG6'));
+    this.route.queryParamMap.pipe(tap(_ => console.log('DEBUG7')));
+
+    this.route.paramMap.pipe(
+      tap(_ => console.log('DEBUG5')),
```

Comment:
Need help for this @kistora.

When I run this, I get
```
DEBUG1
DEBUG2
DEBUG3
DEBUG6
```

I can't get `.pipe` working :cry:

Reply:
You forget to subscribe to you beautiful obversable :)